numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -310,6 +310,10 @@ def bind(self):
         """
         cufunc = self._codelibrary.get_cufunc()
 
+        mod = cufunc.module
+        if not mod.initialized:
+            mod.setup()
```

Comment:
Is this thread safe? Can this accidentally get run multiple times and cause unintended behavior?

Reply:
Are we discussing about the use of multitheading in free-threaded python scenario? If multiple threads launches the kernel, each thread can indeed see the `initialized` as `False` and therefore run `setup` multiple times.

In general, do we want to start considering for free-threaded python for all new features?

Reply:
No and yes, I think.

Reply:
(didn't see the other replies between starting and finishing thinking about this - I was replying to Keith)

Reply:
Discussed offline that we can go with using a lock to make sure thread safety. If there comes another ask to lift such restriction from another library, we'll revisit.